butler/src/test/java/rocks/metaldetector/butler/client/ReleaseButlerRestClientImplTest.java

```diff
@@ -557,4 +559,75 @@ private Stream<Arguments> httpStatusCodeProvider() {
       return Stream.of(HttpStatus.values()).filter(status -> !status.is2xxSuccessful()).map(Arguments::of);
     }
   }
+
+  @DisplayName("Tests getting statistics")
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  @Nested
+  class StatisticsTest {
+
+    @Test
+    @DisplayName("restTemplate is called with correct url")
+    void test_rest_template_called() {
+      // given
+      var url = "url";
+      doReturn(url).when(butlerConfig).getStatisticsUrl();
+      doReturn(ResponseEntity.ok(ButlerStatisticsResponse.builder().build())).when(restTemplate).getForEntity(anyString(), eq(ButlerStatisticsResponse.class));
+
+      // when
+      underTest.getStatistics();
+
+      // then
+      verify(restTemplate).getForEntity(eq(url), eq(ButlerStatisticsResponse.class));
+    }
+
+    @Test
+    @DisplayName("response is returned")
+    void test_response_returned() {
+      // given
+      var expectedResponse = ButlerStatisticsResponse.builder()
+          .releaseInfo(ButlerReleaseInfo.builder().build())
+          .build();
+      doReturn("url").when(butlerConfig).getStatisticsUrl();
+      doReturn(ResponseEntity.ok(expectedResponse)).when(restTemplate).getForEntity(anyString(), eq(ButlerStatisticsResponse.class));
+
+      // when
+      var result = underTest.getStatistics();
+
+      // then
+      assertThat(result).isEqualTo(expectedResponse);
+    }
+
+    @Test
+    @DisplayName("if the status code is not OK on getting statistics, an ExternalServiceException is thrown")
+    void test_exception_response_null() {
+      // given
+      doReturn("url").when(butlerConfig).getStatisticsUrl();
+      doReturn(ResponseEntity.ok().build()).when(restTemplate).getForEntity(anyString(), eq(ButlerStatisticsResponse.class));
+
+      // when
+      Throwable throwable = catchThrowable(() -> underTest.getStatistics());
+
+      // then
+      assertThat(throwable).isInstanceOf(ExternalServiceException.class);
+    }
+
+    @ParameterizedTest(name = "if the status is {0}, an ExternalServiceException is thrown")
+    @MethodSource("httpStatusCodeProvider")
+    @DisplayName("if the status code is not OK on getting statistics, an ExternalServiceException is thrown")
+    void test_statistics_exception_if_status_is_not_ok(HttpStatus httpStatus) {
+      // given
+      doReturn("url").when(butlerConfig).getStatisticsUrl();
+      doReturn(ResponseEntity.status(httpStatus).body(ButlerStatisticsResponse.builder().build())).when(restTemplate).getForEntity(anyString(), eq(ButlerStatisticsResponse.class));
+
+      // when
+      Throwable throwable = catchThrowable(() -> underTest.getStatistics());
+
+      // then
+      assertThat(throwable).isInstanceOf(ExternalServiceException.class);
+    }
+
+    private Stream<Arguments> httpStatusCodeProvider() {
```

Comment:
and now use private static classes in inner classes. The TestInstance workaround is no longer necessary.

```suggestion
    private static Stream<Arguments> httpStatusCodeProvider() {
```